### DISCUSSION TITLE
Stupid questions II: Electric Boogaloo: So, how do I get a the list of servers a user can see?

### DISCUSSION THREAD

**Discussion by alexispurslane** [stoatchat/discussions#608] [Q&A] [answered]
Been pouring through the source code for revolt.js and rvmob and haven't found what API call I need to make to get the list of servers that the signed-in user can see. I've looked at revolt.js's `Server.ts` and `Client.ts`, and rvmob's `ServerList.tsx`, `SideMenus.tsx`, and `App.tsx`. I can see where RVMob is pulling an ordered list of servers from the user's settings, but that doesn't exist yet for my user. I can see where it's calling `client.servers.values()`, but I can't figure out what's actually turning `client.servers` into a list of servers, because running over the code in revolt.js, there's no obvious point where it makes any sort of request that would produce that list. I'm very sleep deprived today so I could be missing something obvious, but I really do need help at this point. Thanks again :sweat:

**Comment by sussycatgirl** [comment, marked as answer]
When you connect to the websocket, you get a `Ready` event which includes servers and relationships. As far as I'm aware, there's no API route to fetch these yet

**Reply by alexispurslane** [reply]
Got it, thanks!
